canvas_modules/common-canvas/src/common-properties/util/property-utils.js

```diff
@@ -183,6 +183,72 @@ function convertType(storage) {
 	return retVal;
 }
 
+/**
+ * Converts the currentValues of a structure control of type 'object'
+ *  from an array of objects to an array of values.
+ *  Example currentValues: [[{a: 1, b: 2}], [{a: 10, b; 20}]]
+ *  Example convertedValues: [[1, 2], [10, 20]]
+ */
+function convertObjectStructureToArray(controlType, subControls, currentValues) {
+	const structureKeys = [];
+	subControls.forEach((control) => {
+		structureKeys.push(control.name);
+	});
+
+	if (controlType === ControlType.STRUCTUREEDITOR) {
```

Comment:
Why do we need to check control type?  The control really shouldn't matter.  It should be base on the type.

Reply:
This function will only be called in the case where `type` equals `object`. This check is needed to differentiate the different currentValues received. `STRUCTUREEDITOR` is a single object since it doesn't have any rows whereas the other stcutures have 1 or many rows. I'll need to fix the example values in the comment